single_iwyu.py

```diff
@@ -22,7 +22,16 @@ def main(commands: Path, fixer_path: Path, filters: List[Path], specific: str):
         if len(eligible) == 0:
             warn("NO FILE WITH IDENTIFIER FOUND")
         for part in eligible:
-            perform_iwyu(fixer_path, part, filters, current_dir, debug=DEBUG)
+            if not "arch" in part["file"]:
+                perform_iwyu(
+                    fixer_path,
+                    part,
+                    filters + [Path(current_dir), "nonarch.imp"],
+                    current_dir,
+                    debug=DEBUG,
+                )
+            else:
+                perform_iwyu(fixer_path, part, filters, current_dir, debug=DEBUG)
```

Comment:
You should make a new var for `filters` and append to it if `"arch" in part["file"]`, or something like:

```python
for part in eligible:
  if not "arch" in part["file"]:
    filters += [Path ...
  perform_iwyu(fixer_path, ...
```

Reply:
won't we only be running single_iwyu.py on one .c file at a time?